extensions/metadata.md

```diff
@@ -0,0 +1,740 @@
+    C: METADATA * UNSUB bar foo baz
+    S: :irc.example.com RPL_METADATAUNSUBOK modernclient :baz foo bar
+    C: METADATA * SUBS
+    S: :irc.example.com RPL_METADATASUBS modernclient :avatar website
+
+#### Subscribing to the same key multiple times 1
+
+    C: METADATA * SUB website avatar foo bar baz
+    S: :irc.example.com RPL_METADATASUBOK modernclient :website avatar foo bar baz
+    C: METADATA * SUBS
+    S: :irc.example.com RPL_METADATASUBS modernclient :avatar bar baz foo website
+    C: METADATA * SUB avatar website
+    S: :irc.example.com RPL_METADATASUBOK modernclient :avatar website
+    C: METADATA * SUBS
+    S: :irc.example.com RPL_METADATASUBS modernclient :avatar bar baz foo website
+
+#### Subscribing to the same key multiple times 2
+
+The client (erroneously) subscribes to the same key twice in the same command.
+The server is free to include the key being subscribed to in the
+`RPL_METADATASUBOK` numeric once or twice.
+
+In both cases, the key will only appear once in the reply to a following
+`METADATA SUBS` command.
+
+Once:
+
+    C: METADATA * SUB avatar avatar
+    S: :irc.example.com RPL_METADATASUBOK modernclient :avatar
+    C: METADATA * SUBS
+    S: :irc.example.com RPL_METADATASUBS modernclient :avatar
+
+Twice:
+
+    C: METADATA * SUB avatar avatar
+    S: :irc.example.com RPL_METADATASUBOK modernclient :avatar avatar
+    C: METADATA * SUBS
+    S: :irc.example.com RPL_METADATASUBS modernclient :avatar
+
+#### Unsubscribing from a non-subscribed key 1
+
+    C: METADATA * SUBS
+    C: METADATA * UNSUB website
+    S: :irc.example.com RPL_METADATAUNSUBOK modernclient :website
+    C: METADATA * SUBS
+    C: METADATA * SUB website
+    S: :irc.example.com RPL_METADATASUBOK modernclient :website
+    C: METADATA * SUBS
+    S: :irc.example.com RPL_METADATASUBS modernclient :website
+
+#### Unsubscribing from a non-subscribed key 2
+
+The client (erroneously) unsubscribes from the same key twice in the same
+command. The server is free to include the key being unsubscribed from in the
+`RPL_METADATAUNSUBOK` numeric once or twice.
+
+Once:
+
+    C: METADATA * SUBS
+    S: :irc.example.com RPL_METADATASUBS modernclient :website
+    C: METADATA * UNSUB website website
+    S: :irc.example.com RPL_METADATAUNSUBOK modernclient :website
+
+Twice:
+
+    C: METADATA * SUBS
+    S: :irc.example.com RPL_METADATASUBS modernclient :website
+    C: METADATA * UNSUB website website
+    S: :irc.example.com RPL_METADATAUNSUBOK modernclient :website website
+
+#### Subscribing to a key which requires privileges but without privileges
+
+    C: METADATA * SUB avatar secretkey website
+    S: FAIL METADATA KEY_NO_PERMISSION secretkey modernclient :You do not have permission to do that.
+    S: :irc.example.com RPL_METADATASUBOK modernclient :secretkey website
+    C: METADATA * SUBS
+    S: :irc.example.com RPL_METADATASUBS modernclient :secretkey website
```

Comment:
These should probably be avatar and website, since secretkey was already rejected